lib/internal/modules/esm/hooks.js

```diff
@@ -152,13 +152,16 @@ class Hooks {
     } = pluckHooks(exports);
 
     if (globalPreload) {
-      ArrayPrototypePush(this.#chains.globalPreload, { fn: globalPreload, url });
+      const next = this.#chains.globalPreload[this.#chains.globalPreload.length - 1];
+      ArrayPrototypePush(this.#chains.globalPreload, { fn: globalPreload, url, next });
```

Comment:
`globalPreload` hooks don’t get chained and don’t get a `next` argument. Perhaps this line shouldn’t have been changed?

```suggestion
      ArrayPrototypePush(this.#chains.globalPreload, { fn: globalPreload, url });
```

Reply:
if these objects won't always have the same three keys, then they should all have `__proto__: null`

Reply:
@ljharb there is no chance of these getting set when all the fields are not set, so the `__proto__: null` isn't be necessary.

Reply:
No, I mean the fields we are setting are for sure set, so there is no opportunity for a field from the prototype to be accidentally reached.

Reply:
right, that's the current case in this PR, which is why i haven't suggested proto null so far - but if this suggestion is accepted, the `next` field is *not* set here, and thus the opportunity manifests.

Reply:
Oh gosh 🤦‍♂️ you're right!

Reply:
ah ok, so globalPreload only accesses these two fields, and resolve/load accesses all three? in that case, the suggestion's fine as-is :-)

(although i think this is all a bit unclear and a potential footgun, and it'd probably be safest to either always use null protos, or, to make sure to only access own properties off of these objects on the other side)

Reply:
> globalPreload only accesses these two fields, and resolve/load accesses all three?

✅ correct 🙂

> this is all a bit unclear and a potential footgun

We're about to replace globalPreload, so the chance of the foot-gun happening (only a concern for node contributors) is basically non-existent.

Reply:
Updated!